### PR TITLE
Topuses

### LINKED ISSUE #1353: top uses button needs some style fixes[UX]
**Describe the bug**
top uses button needs some style fixes

**Screenshots**
users

**Are you working on this issue? (Yes/No)**
YES

### PR DESCRIPTION
**Pull request by mahmodHammad**
Fixes #1353 

#### Describe the changes you have made in this PR -
- adjust padding 
- fix text color on hover 
- add a border to the top users btn
#### AFTER
bbbafters
#### BEFORE

bbbefore

### CODE REVIEW

**Comment by yash15601** [comment]
@mahmodHammad  don't override the color make new color variable  in theme.scss  and define your color and use that

**Comment by mahmodHammad** [comment]
@yash15601 
what do you mean !?

**Comment by yash15601** [comment]
> @yash15601
> what do you mean !?

dont use !important inshort define new color in theme.scss and use it there

**Comment by mahmodHammad** [comment]
> > @yash15601
> > what do you mean !?
> 
> dont use !important inshort define new color in theme.scss and use it there

$white is already defined
Could you explain how this could work without. !Important?

File: app/assets/stylesheets/examples.scss [hunk 1]
```diff
@@ -49,14 +50,15 @@
...
+  border: 1px solid $green;
   border-radius: 30px;
-  color: $white;
   transition: all .3s ease 0s;
 
   &:hover {
     background-color: $teal;
     box-shadow: 0 8px 15px $shadow-black;
-    color: $white;
+    color: $white !important;
```
[the remark this answers is not in the document]
Comment (mahmodHammad) [reply app/assets/stylesheets/examples.scss:61]: I found no thing else that can solve it without important...if any one have a solution, please share that with us

**Comment by yash15601** [comment]
> > > @yash15601
> > > what do you mean !?
> > 
> > 
> > dont use !important inshort define new color in theme.scss and use it there
> 
> $white is already defined
> Could you explain how this could work without. !Important?

go to assest/css/theme.css
define color $xyz="#yourcolor"
 then instead of using  $White use that if white is already define then why you are overriding color it can effect to an part of ui if that class is used

**Comment by mahmodHammad** [comment]
> > > > @yash15601
> > > > what do you mean !?
> > > 
> > > 
> > > dont use !important inshort define new color in theme.scss and use it there
> > 
> > 
> > $white is already defined
> > Could you explain how this could work without. !Important?
> 
> go to assest/css/theme.css
> define color $xyz="#yourcolor"
> then instead of using $White use that if white is already define then why you are overriding color it can effect to an part of ui if that class is used

again please check the code

**Comment by mahmodHammad** [comment]
> > > > @yash15601
> > > > what do you mean !?
> > > 
> > > 
> > > dont use !important inshort define new color in theme.scss and use it there
> > 
> > 
> > $white is already defined
> > Could you explain how this could work without. !Important?
> 
> go to assest/css/theme.css
> define color $xyz="#yourcolor"
> then instead of using $White use that if white is already define then why you are overriding color it can effect to an part of ui if that class is used

whatever we do it will sill black without !important.

why do you think it can effect to an part of ui if that class is used?!

File: app/assets/stylesheets/examples.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (aayush-05) [reply app/assets/stylesheets/examples.scss:61]: This is basically due to some conflicting styles, may be we can continue this PR after some stylesheet cleanup.

**Comment by abstrekt** [comment]
> 
> 
> > > > > @yash15601
> > > > > what do you mean !?
> > > > 
> > > > 
> > > > dont use !important inshort define new color in theme.scss and use it there
> > > 
> > > 
> > > $white is already defined
> > > Could you explain how this could work without. !Important?
> > 
> > 
> > go to assest/css/theme.css
> > define color $xyz="#yourcolor"
> > then instead of using $White use that if white is already define then why you are overriding color it can effect to an part of ui if that class is used
> 
> whatever we do it will for some reason sill black without !important.
> if you can fix it without !important go-ahead
conflicting styles,
put an ID in the target element & set the color property to that ID, ID has the highest specificity.

**Comment by yash15601** [comment]
Code climate will give another  issue avoid using  selector

**Comment by mahmodHammad** [comment]
> > > > > > @yash15601
> > > > > > what do you mean !?
> > > > > 
> > > > > 
> > > > > dont use !important inshort define new color in theme.scss and use it there
> > > > 
> > > > 
> > > > $white is already defined
> > > > Could you explain how this could work without. !Important?
> > > 
> > > 
> > > go to assest/css/theme.css
> > > define color $xyz="#yourcolor"
> > > then instead of using $White use that if white is already define then why you are overriding color it can effect to an part of ui if that class is used
> > 
> > 
> > whatever we do it will for some reason sill black without !important.
> > if you can fix it without !important go-ahead
> 
> * conflicting styles,
>   put an ID in the target element & set the color property to that ID, ID has the highest specificity.

What if we wanted to make an other button, with the same style.. It is not possible to give multiple element the same id

**Comment by mahmodHammad** [comment]
I only put .!important  on the hover effect, I don't see that as a terrible think to avoid, because we only override the color on hover

**Comment by mahmodHammad** [comment]
Of it's  a big issue, can make only make  inline style in the htmlwhich is bad solutions too,

**Comment by abstrekt** [comment]
then close this is PR if you can't solve the issue.

**Comment by mahmodHammad** [comment]
> then close this is PR if you can't solve the issue.

!!!

**Comment by yash15601** [comment]
In which book it is written  that if you cant solve it close  the issue he is trying  will came with new solution

**Comment by yash15601** [comment]
As aayush and nitin said  earlier  their will rules for button to maintain that time I think this will be fixed

**Comment by abstrekt** [comment]
> In which book it is written that if you cant solve it close the issue he is trying will came with new solution

www.github.com/PR/Rule_Book-101

**Comment by abstrekt** [comment]
@mahmodHammad  @yash15601 
The text color on hover is applied by bootsrap with this selector ```a:not([href]):not([tabindex]):hover```, now since class couldn't override that (they usually do), that means this selector got higher specificity, thus to get the desired result one must use selector with higher specificity than this, which leaves him with 3 options, over-ride the bootstrap global selector & turn every acher tag on hover to white, else use an ID or inline-style, now if he doesn't want to use ID or inline style there is no solution to this **particular specificity problem**.

**Comment by yash15601** [comment]
We can make new class   for that specific  button and then we go further that is the one simple solution  I know their will no error by code  climate

**Pull request merged**